src/dictionary.c

```diff
@@ -70,8 +220,13 @@ static void dict_init_impl(struct dictionary* d, SEXP x, bool partial) {
   } else {
     d->hash = NULL;
   }
+
+  UNPROTECT(1);
+  return d;
 }
 
+
+// Use hash from `x` but value from `d`?
```

Comment:
This was for partial dictionary support, like `vec_match()` and `vec_in()` right?